backend/services/implementations/testSessionService.ts

```diff
@@ -189,6 +204,69 @@ class TestSessionService implements ITestSessionService {
 
     return testSessionDtos;
   }
+
+  /*
+   * computeTestGrades computes the breakdown and score of a given
+   * UngradedResultDTO and returns the GradedResultResponseDTO
+   */
+  async computeTestGrades(
+    result: UngradedResultDTO,
+    testId: string,
+  ): Promise<GradedResultResponseDTO> {
+    let gradedResultResponseDTO: GradedResultResponseDTO;
+
+    // the list of a student's answers with each field being either the
+    // numeric answer (for short answer) or index (for multiple choice)
+    const studentAnswers: (number | null)[] = result.answers;
+
+    let computedScore = 0.00;
+    const computedBreakdown: boolean[] = [];
+    let questionsCorrect = 0;
+
+    try {
+      const test: TestResponseDTO = await this.testService.getTestById(testId);
+
+      test.questions.forEach((question: Question, i) => {
+        let actualAnswer: number;
+
+        if (question.questionType === QuestionType.MULTIPLE_CHOICE) {
+          const questionMetadata = question.questionMetadata as MultipleChoiceMetadata;
+          actualAnswer = questionMetadata.answerIndex;
+        } else {
+          const questionMetadata = question.questionMetadata as NumericQuestionMetadata;
+          actualAnswer = questionMetadata.answer;
+        }
```

Comment:
we can move this entire block into a helper function called `getCorrectAnswer` or something